dotnet/src/Agents/Core/AgentGroupChat.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Agents.Chat;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// A an <see cref="AgentChat"/> that supports multi-turn interactions.
+/// </summary>
+public sealed class AgentGroupChat : AgentChat
+{
+    private readonly HashSet<string> _agentIds; // Efficient existence test
```

Comment:
The intent here is to avoid a linear look-up when accessing the agent collection.  While `Dictionary<string, Agent>` would accomplish this, the desire to maintain ordering is defeated due to the nature of the dictionary/hashing.

> Aside: I've often pondered an "ordered" dictionary...but because of the nature of list vs hash (both construction and consumption), you still need two separate containers.